src/App.js

```diff
@@ -1,21 +1,103 @@
-import React, { useState, useEffect } from 'react';
-import PokemonList from './PokemonList';
-import axios from "axios";
-import './App.css';
+import React, { useState, useEffect } from "react";
+import PokemonList from "./PokemonList";
+import Axios from "axios";
+import "./App.css";
+import PaginationFunction from "./Pagination";
+import Spinner from 'react-bootstrap/Spinner'
+import Limit from "./selectLimit";
+import pokeBall from './Images/pokeBall.png'
 
 function App() {
-  const [pokemon, setPokemon] = useState([]);
+  const [pageLimit, setPageLimit] = useState(20);
+  const [currentPage, setcurrentPage] = useState(1);
+  const offset = (currentPage - 1) * pageLimit;
+  const url = `${process.env.REACT_APP_POKE}?offset=${offset}&limit=${pageLimit}`;
+  const [totalPages, settotalPages] = useState(1);
+  const [data, setData] = useState([]);
+  const [loading, setloading] = useState(true);
 
   useEffect(() => {
-    axios.get("https://pokeapi.co/api/v2/pokemon").then(res => {
-      setPokemon(res.data.results.map(p => p.name))
-    })
+    setloading(true);
+    fetchPokemonList();
+    setloading(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentPage, pageLimit]);
 
-  }, [])
+  const fetchPokemonList = async () => {
+    let cancel;
+    await Axios
+      .get(url, { cancelToken: new Axios.CancelToken((c) => (cancel = c)) })
+      .then((res) => {
+        settotalPages(Math.ceil(res.data.count / pageLimit));
+        let useData = (res.data.results.map((u)=> u.url));
+        fetchPokemonData(useData);
+      });
 
+    return () => cancel();
+  };
+
+  const fetchPokemonData = async (u) => {
+    const promise = u.map(async (items) => {
+      const response = await Axios.get(items);
+      let name = response.data.name;
+      let formalName = name[0].toUpperCase() + name.substring(1);
+      const height = response.data.height * 3.9;
+      const heightConvert = Math.round(height * 10) / 10
+      const weight = response.data.weight / 4.5;
+      const weightConvert = Math.round(weight * 10) / 10
+
+      return {
+        name: formalName,
+        id: response.data.id,
+        weight: `${weightConvert} lbs`,
+        height: `${heightConvert} Inches`,
+        sprite: response.data.sprites.front_default,
+        type: response.data.types.map((n) => {
+          let hold = n.type.name;
+          return hold[0].toUpperCase() + hold.substring(1);
+        }),
```

Comment:
this whole block 42-58line  where you format data should be separate inside
Pokemon.helpers.js or PokemonUtils.js files...

Basically the rule is that whenever you have huge block of code like this it is extremely hard to read,
and if you put it separate then it will be all inside `getFormattedPokemonData` function which is easier to read

Reply:
created a separate function to format the inputted array of URLs and return back a formatted object of only necessary information from the API request